osu.Game/Graphics/UserInterface/PolyBar.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osuTK;
+using osuTK.Graphics;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    // A modified implementation of Bar which supports multiple accent bars
+    public partial class PolyBar : Container
+    {
+        // Box to represent the background of the bars
+        private readonly Box background;
+
+        // Each accent coloured bar within the PolyBar
+        private readonly Box[] bars;
+
+        // number of bars, could use bars.Length instead but moved to readonly field for readability
+        private readonly int barCount;
+
+        private const int resize_duration = 250;
+
+        private const Easing easing = Easing.InOutCubic;
+
+        // Each of these maps to a box in bars 1:1, lengths[i] represents the length of bars[i]
+        private readonly float[] lengths;
+
+        // getters and setters for the lengths are moved to functions, because C# doesn't support indexed getters
+        public void SetLength(int index, float value)
+        {
+            lengths[index] = Math.Clamp(value, 0, 1);
+            updateBarLength(index);
+        }
```

Comment:
This was just confusion in language, indexers don't work here because its accessing two arrays in the class, and it's more readable to have the separate get/set functions

Reply:
I would rather avoid java-style getters/setters. `BindableList` is a possible replacement. Or just a proper class-based API wherein you can get a single `Bar` from this thing and set things on it directly.